### DISCUSSION TITLE
[Console] broken configuration after saving on v9.5.1

### DISCUSSION THREAD

**Discussion by Noste92** [mia-platform/community#44] [Q&A] [answered]
Hi,

I've noticed a strange behavior working on the console v9.5.1

I've changed the timeout of the API gateway, after saving these changed happened on configuration:

- A new file locations.conf has been generated with all the location configured in the endpoints and all the location includes the root-location-extension.conf previously configured.
- this locations.conf has been included in the serverConf.conf

The problem is that root-location-extension.conf contains a configuration on a target location, including this configuration in other locations breaks the configuration, here's a series of example of files before and after the update.

BEFORE

serverConf.conf
```
include /etc/nginx/customization.d/rate-limit-usage.conf;
include /etc/nginx/customization.d/auth-usage.conf;
include /etc/nginx/customization.d/auth-location.conf;
include /etc/nginx/customization.d/server-extension.conf;
include /etc/nginx/customization.d/default-error-page.conf;

set $isbackoffice "";

location / {
  include /etc/nginx/customization.d/header-debug.conf;
  include /etc/nginx/customization.d/auth-set-userid.conf;
  include /etc/nginx/set_proxy_headers.conf;
  include /etc/nginx/customization.d/root-location-extension.conf;

  proxy_pass http://$proxy_name$proxy_url;
}
```

root-location-extension.conf
```
proxy_set_header 'secret' $client_type;

location /my/custom/location {
    client_max_body_size 5m; //Previously configured here instead of specific console sections
    proxy_pass http://my-server/my/custom/location;
}
```

AFTER

serverConf.conf
```
include /etc/nginx/customization.d/rate-limit-usage.conf;
include /etc/nginx/customization.d/auth-usage.conf;
include /etc/nginx/customization.d/auth-location.conf;
include /etc/nginx/customization.d/server-extension.conf;
include /etc/nginx/customization.d/default-error-page.conf;

set $isbackoffice "";

include /etc/nginx/customization.d/locations.conf;

location / {
  include /etc/nginx/customization.d/header-debug.conf;
  include /etc/nginx/customization.d/auth-set-userid.conf;
  include /etc/nginx/set_proxy_headers.conf;
  include /etc/nginx/customization.d/root-location-extension.conf;

  proxy_pass http://$proxy_name$proxy_url;
}
```

root-location-extension.conf
```
proxy_set_header 'secret' $client_type;

location /my/custom/location {
    client_max_body_size 5m; //Previously configured here instead of specific console sections
    proxy_pass http://my-server/my/custom/location;
}
```

locations.conf

```conf
  location ~ ^/another/location {
    include /etc/nginx/customization.d/header-debug.conf;
    include /etc/nginx/customization.d/auth-set-userid.conf;
    include /etc/nginx/set_proxy_headers.conf;
    include /etc/nginx/customization.d/root-location-extension.conf; // this include breaks the configuration

    proxy_pass http://$proxy_name$proxy_url;
  }

  location ~ ^/my/custom/location {
    include /etc/nginx/customization.d/header-debug.conf;
    include /etc/nginx/customization.d/auth-set-userid.conf;
    include /etc/nginx/set_proxy_headers.conf;
    include /etc/nginx/customization.d/root-location-extension.conf;

    proxy_pass http://$proxy_name$proxy_url;
  }
```

I could resolved this problem by configure the client_max_body_size directly in the console and removing location config in the root-location-extension.conf file, but i had to figure out the solution.

What do you think about this?

**Comment by davidebianchi** [comment, marked as answer]
Hi @Noste92, do you have set a custom timeout in one of the endpoints defined on Console?
What type of configuration error raise nginx?

I suppose the Console, if the timeout is not changed in at least one endpoint, should not change the previous configuration.

**Reply by Noste92** [reply]
Hi @davidebianchi,

I have updated a custom timeout under Advanced => api-gateway => rate-limit-usage.conf inferred from a public variable, here's the content

```conf
proxy_read_timeout {{API_GATEWAY_TIMEOUT_SECONDS}}s;
```

Nginx has raised this error:
```
[emerg] 1#1: location "/v1/services/mail/send" is outside location "^/v2/ip-blacklist" in /etc/nginx/customization.d/root-location-extension.conf:3
```

**Reply by fredmaggiowski** [reply]
Hi @Noste92 just to be sure: do you have any timeout set in any endpoint detail page? (not from advanced)

**Reply by Noste92** [reply]
Hi @fredmaggiowski if you mean the property timeout on endpoints => endpoint settings => generals, I have checked the endpoints in the project but no, I don't have any timeout set there.

**Reply by davidebianchi** [reply]
We have now released (v9.5.3 of the Console) a rollback of the feature for the nginx api-gateway, to fix the previous behaviour. This should solve this issue, in the next release we would understand how to give the feature in a fully compatible way.
